File: frontend/src/lib/types/medication.ts
export interface Medication {
	id: number;
	owner: string;
	name: string;
	dosage: string;
	unit: string;
	frequency: string;
	purpose?: string;
	start_date: string;
	end_date?: string;
	prescribing_doctor?: string;
	refill_reminder_date?: string;
	is_active: boolean;
	created_at: string;
	updated_at: string;
}

export interface MedicationCreate {
	owner: string;
	name: string;
	dosage: string;
	unit: string;
	frequency: string;
	purpose?: string;
	start_date: string;
	end_date?: string;
	prescribing_doctor?: string;
	refill_reminder_date?: string;
}

export interface MedicationUpdate {
	name?: string;
	dosage?: string;
	unit?: string;
	frequency?: string;
	purpose?: string;
	start_date?: string;
	end_date?: string;
	prescribing_doctor?: string;
	refill_reminder_date?: string;
	is_active?: boolean;
}

File: backend/app/schemas/medication.py
from datetime import date, datetime

from pydantic import BaseModel, Field


class MedicationBase(BaseModel):
    name: str = Field(..., max_length=200)
    dosage: str = Field(..., max_length=100)
    unit: str = Field(..., max_length=50)
    frequency: str = Field(..., max_length=100)
    purpose: str | None = None
    start_date: date
    end_date: date | None = None
    prescribing_doctor: str | None = Field(None, max_length=200)
    owner: str = Field(..., max_length=100)
    is_active: bool = True
    refill_reminder_date: date | None = None


class MedicationCreate(MedicationBase):
    pass


class MedicationUpdate(BaseModel):
    name: str | None = Field(None, max_length=200)
    dosage: str | None = Field(None, max_length=100)
    unit: str | None = Field(None, max_length=50)
    frequency: str | None = Field(None, max_length=100)
    purpose: str | None = None
    start_date: date | None = None
    end_date: date | None = None
    prescribing_doctor: str | None = Field(None, max_length=200)
    owner: str | None = Field(None, max_length=100)
    is_active: bool | None = None
    refill_reminder_date: date | None = None


class MedicationResponse(MedicationBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
